homeassistant/components/withings/config_flow.py

```diff
@@ -0,0 +1,209 @@
+"""Config flow for Withings."""
+from collections import OrderedDict
+import logging
+from typing import Optional
+
+import aiohttp
+import nokia
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+
+from . import const
+
+DATA_FLOW_IMPL = 'withings_flow_implementation'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def register_flow_implementation(
+        hass,
+        client_id,
+        client_secret,
+        base_url,
+        profiles
+):
+    """Register a flow implementation.
+
+    domain: Domain of the component responsible for the implementation.
+    name: Name of the component.
+    client_id: Client id.
+    client_secret: Client secret.
+    """
+    if DATA_FLOW_IMPL not in hass.data:
+        hass.data[DATA_FLOW_IMPL] = OrderedDict()
+
+    hass.data[DATA_FLOW_IMPL] = {
+        const.CLIENT_ID: client_id,
+        const.CLIENT_SECRET: client_secret,
+        const.BASE_URL: base_url,
+        const.PROFILES: profiles
+    }
+
+
+@config_entries.HANDLERS.register(const.DOMAIN)
+class WithingsFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self.flow_profile = None
+        self.data = None
+
+    def async_profile_config_entry(
+            self,
+            profile: str
+    ) -> Optional[ConfigEntry]:
+        """Get a profile config entry."""
+        entries = self.hass.config_entries.async_entries(const.DOMAIN)
+        for entry in entries:
+            if entry.data.get(const.PROFILE) == profile:
+                return entry
+
+        return None
+
+    def get_auth_client(self, profile: str):
+        """Get a new auth client."""
+        flow = self.hass.data[DATA_FLOW_IMPL]
+        client_id = flow[const.CLIENT_ID]
+        client_secret = flow[const.CLIENT_SECRET]
+        base_url = flow[const.BASE_URL].rstrip('/')
+
+        callback_uri = '{}/{}?flow_id={}&profile={}'.format(
+            base_url.rstrip('/'),
+            const.AUTH_CALLBACK_PATH.lstrip('/'),
+            self.flow_id,
+            profile
+        )
+
+        return nokia.NokiaAuth(
+            client_id,
+            client_secret,
+            callback_uri,
+            scope=','.join(['user.info', 'user.metrics', 'user.activity'])
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Create an entry for selecting a profile."""
+        flow = self.hass.data.get(DATA_FLOW_IMPL, {})
+
+        if user_input and const.PROFILE in user_input:
```

Comment:
The profile is required in the form schema so we shouldn't need to check for that here.

Reply:
This is necessary as to short-circuit the config flow to goto the auth flow if the profile is provided.

Reply:
The profile will always be provided if any user input is provided.

Reply:
I'm a bit confused on what's being requested here. The check for the profile is necessary to advance the flow to the "auth" step. This may be based on a lack of understanding config flows. This is how I understand the config flow process to work:
- Config flow gets "user" step, which returns a show form step with the profile list.
- User selects a profile from the list.
- Config flow gets "user" step with the user's input. Which returns the external step with id of "auth".

Reply:
This is enough.
```py
if user_input:
```